ethereumj-core/src/main/java/org/ethereum/vm/DataWord.java

```diff
@@ -81,18 +86,26 @@ public DataWord(byte[] data) {
             this.data = ByteUtil.EMPTY_BYTE_ARRAY;
         else if (data.length == 32)
             this.data = data;
-        else if (data.length <= 32)
-            System.arraycopy(data, 0, this.data, 32 - data.length, data.length);
-        else
-            throw new RuntimeException("Data word can't exceed 32 bytes: " + data);
+        else if (data.length <= 32) {
+            byte[] bytes = ZERO.getData();
+            System.arraycopy(data, 0, bytes, 32 - data.length, data.length);
+            this.data = bytes;
+        } else
+            throw new RuntimeException(String.format("Data word can't exceed 32 bytes: 0x%s", ByteUtil.toHexString(data)));
     }
 
     public byte[] getData() {
-        return data;
+        return Arrays.copyOf(data, data.length);
+    }
```

Comment:
I would leave this as is, just making sure the returned array is not accidentally modified else where (which normally shouldn't be the case for `getData()` method). If one really needs a copy I'd suggest add special `getDataCopy()` method